### PR TITLE
ipn/ipnlocal,tailcfg: add AppConnector service to HostInfo when configured

### PR DESCRIPTION
**Pull request by raggi**
Updates tailscale/corp#15437

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -3171,6 +3171,12 @@ func (b *LocalBackend) peerAPIServicesLocked() (ret []tailcfg.Service) {
 			Port:  1, // version
 		})
 	}
+	if b.appConnector != nil {
+		ret = append(ret, tailcfg.Service{
+			Proto: tailcfg.AppConnector,
+			Port:  1, // version
```
Comment (twitchyliquid64) [inline ipn/ipnlocal/local.go:3177]: Add this to the docstring on `Service.Proto` in the existing style

Comment (raggi) [reply ipn/ipnlocal/local.go:3177]: done, though I think they should move near to the constants to scale better over time

**Comment by twitchyliquid64** [comment]
Quick rebase/deconflict so i can do another round of review?

**Review by creachadair — APPROVED** [review]
The change LGTM to me, but the build seems genuinely unhappy.

**Pull request merged**